lib/atom/decoration.js

```diff
@@ -64,12 +64,12 @@ class BareDecoration extends React.Component {
   componentDidUpdate(prevProps) {
     if (this.props.editorHolder !== prevProps.editorHolder) {
       this.editorSub.dispose();
-      this.editorSub = this.state.editorHolder.observe(this.observeParents);
```

Comment:
did you change from `state` to `props` because that's easier to test?

Reply:
No, it was actually just broken before 😅 We hadn't had any tests that covered an update of an existing component, and apparently it doesn't come up in our package yet. But `editorHolder` is a member of the state in the wrapper class that reads Context, not in the `BareDecoration`!